src/ui/zabapgit_js_common.w3mi.data.js

```diff
@@ -1407,3 +1438,253 @@ function GitGraphScroller() { // eslint-disable-line no-unused-vars
+    var newCmd = this.commands[newIndex];
+    newCmd.element.classList.add("selected");
+    this.selectIndex = newIndex;
+    this.adjustScrollPosition(newCmd.element);
+  }
+};
+
+CommandPalette.prototype.selectFirst = function(){
+  for (var i = 0; i < this.commands.length; i++) {
+    if (this.commands[i].element.style.display === "none") continue; // skip hidden
+    this.applySelectIndex(i);
+    break;
+  }
+};
+
+CommandPalette.prototype.selectNext = function(){
+  for (var i = this.selectIndex + 1; i < this.commands.length; i++) {
+    if (this.commands[i].element.style.display === "none") continue; // skip hidden
+    this.applySelectIndex(i);
+    break;
+  }
+};
+
+CommandPalette.prototype.selectPrev = function(){
+  for (var i = this.selectIndex - 1; i >= 0; i--) {
+    if (this.commands[i].element.style.display === "none") continue; // skip hidden
+    this.applySelectIndex(i);
+    break;
+  }
+};
+
+CommandPalette.prototype.getSelected = function(){
+  return this.commands[this.selectIndex];
+};
+
+CommandPalette.prototype.adjustScrollPosition = function(itemElement){
+  var bItem         = itemElement.getBoundingClientRect();
+  var bContainer    = this.elements.ul.getBoundingClientRect();
+  bItem.top         = Math.round(bItem.top);
+  bItem.bottom      = Math.round(bItem.bottom);
+  bItem.height      = Math.round(bItem.height);
+  bItem.mid         = Math.round(bItem.top + bItem.height / 2);
+  bContainer.top    = Math.round(bContainer.top);
+  bContainer.bottom = Math.round(bContainer.bottom);
+
+  if ( bItem.mid > bContainer.bottom - 2 ) {
+    this.elements.ul.scrollTop += bItem.bottom - bContainer.bottom;
+  } else if ( bItem.mid < bContainer.top + 2 ) {
+    this.elements.ul.scrollTop += bItem.top - bContainer.top;
+  }
+};
+
+CommandPalette.prototype.toggleDisplay = function(forceState) {
+  var isDisplayed = (this.elements.palette.style.display !== "none");
+  var tobeDisplayed = (forceState !== undefined) ? forceState : !isDisplayed;
+  this.elements.palette.style.display = tobeDisplayed ? "" : "none";
+  if (tobeDisplayed) {
+    this.elements.input.value = "";
+    this.elements.input.focus();
+    this.applyFilter();
+    this.selectFirst();
+  }
+};
+
+CommandPalette.prototype.getCommandByElement = function(element) {
+  for (var i = 0; i < this.commands.length; i++) {
+    if (this.commands[i].element === element) return this.commands[i];
+  }
+};
+
+CommandPalette.prototype.handleUlClick = function(event) {
+  var element = event.target || event.srcElement;
+  if (!element) return;
+  if (element.nodeName === "SPAN") element = element.parentNode;
+  if (element.nodeName === "I") element = element.parentNode;
+  if (element.nodeName !== "LI") return;
+  this.exec(this.getCommandByElement(element));
+};
+
+CommandPalette.prototype.exec = function(cmd) {
+  if (!cmd) return;
+  this.toggleDisplay(false);
+  submitSapeventForm(null, cmd.action);

```

Comment:
What if we have another action than sapevent? E.g. jump to file uses plain js function.

Reply:
Maybe something like this?
https://github.com/sbcgua/abapGit/blob/6f46aa73892383bd80d98012c2d649bf5ec5eef3/src/ui/zabapgit_js_common.w3mi.data.js#L1669-L1673

Reply:
Yes, I though about it but was lazy and decided to postpone
and now I see you implemented that in your jump CP :)
P.S. I just suggest not passing title to the action and `bind` it in enumerator instead - I'll leave the comment in your commit

Reply:
Thx, fixed it in my branch.